perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -346,48 +355,105 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
+  start_gap, stop_gap = analyze.StreamStartAndEndGaps(
+      records['start_time'], records['latency'],
+      all_streams_start, all_streams_duration)
+  if ((start_gap + stop_gap) / all_streams_duration <
+      MULTISTREAM_STREAM_GAP_THRESHOLD):
+    logging.info(
+        'First stream started %s seconds before last stream started', start_gap)
+    logging.info(
+        'Last stream ended %s seconds after first stream ended', stop_gap)
+  else:
+    logging.warning(
+        'Difference between first and last stream start/end times was %s and '
+        '%s, which is more than %s of the benchmark time %s.',
+        start_gap, stop_gap, MULTISTREAM_STREAM_GAP_THRESHOLD,
+        all_streams_duration)
+    metadata['stream_gap_above_threshold'] = True
+
+  records_in_interval = records[
+      analyze.FullyInInterval(records['start_time'],
+                              records['latency'],
+                              all_streams_start,
+                              all_streams_duration)]
+
   # Don't publish the full distribution in the metadata because doing
   # so might break regexp-based parsers that assume that all metadata
   # values are simple Python objects. However, do add an
   # 'object_size_B' metadata field even for the full results because
   # searching metadata is easier when all records with the same metric
   # name have the same set of metadata fields.
-  overall_metadata['object_size_B'] = 'distribution'
+  distribution_metadata = metadata.copy()
+  distribution_metadata['object_size_B'] = 'distribution'
+
+  latency_prefix = 'Multi-stream %s latency ' % operation
   logging.info('Processing %s multi-stream %s results for the full '
-               'distribution.', len(records), operation)
-  _AppendPercentilesToResults(
-      results,
-      (record['latency'] for record in records),
-      metric_name,
+               'distribution.', len(records_in_interval), operation)
+  analyze.AppendStatsAsSamples(
+      records_in_interval['latency'],
       'sec',
-      overall_metadata)
-
+      results,
+      name_prefix=latency_prefix,
+      metadata=distribution_metadata)
+
+  logging.info('Processing %s multi-stream %s results for net throughput',
+               len(records), operation)
+  net_thpt, thpt_with_overhead, total_overhead, overhead_prop = (
+      analyze.AllStreamsThroughputStats(
+          records_in_interval['latency'],
+          records_in_interval['size'],
+          records_in_interval['stream_num'],
+          num_streams,
+          all_streams_duration))
+  logging.info('Benchmark overhead was %s percent of total benchmark time',
+               overhead_prop * 100.0)
+
+  results.extend([
+      sample.Sample(
+          'Multi-stream %s net throughput' % operation,
+          net_thpt, 'B/sec', metadata=distribution_metadata),
+      sample.Sample(
+          'Multi-stream %s net throughput (experimental)' % operation,
```

Comment:
I believe the "experimental" throughput reported here would be most likely lower than the other non experimental one. but the key question is which one is more useful and accurate to the user?

I have an idea that we should try to validate further: when you do your end-to-end tests on this PR, turn on dstat on a separate window that covers the entire duration of your end-to-end test. configure dstat to report network in/out throughput every second, and make sure to redirect the dstat output to a file. Use a resonable object size (distribution) to begin with, such as 10MB or 100MB objects. The numbers from dstat reflects actual user experience in terms of throughput and can be used to validate which of these two ways of calculation is more accurate.